### PR TITLE
Link_for_each_method, Delay_filter, Theme

### PR DESCRIPTION
**Pull request by mazurkevichandrei**

### CODE REVIEW

**Review by bjatta — APPROVED** [review]

File: methods-react/src/App.js [hunk 1]
```diff
@@ -16,33 +21,49 @@ import {
...
-
     const [allMethods, setAllMethods] = useState(Object.getOwnPropertyNames(Array.prototype).map((item,index) =>({id:index+1,name:item,type:LIST_TYPES.MAIN})));
     
+    const [isDayTheme, setIsDayTheme] = useState('true')
+    const theme = isDayTheme ? themesList.dayTheme : themesList.nightTheme
+    
+    const changeTheme = (evt) =>{
+        setIsDayTheme(!isDayTheme);
+        evt.target.textContent = evt.target.textContent === THEME_BUTTON_NAME.BUTTON_TO_DAY 
+        ? THEME_BUTTON_NAME.BUTTON_TO_NIGHT
```
Comment (bjatta) [inline methods-react/src/App.js]: Читер!
А если апользователь руками изменит!!!

File: methods-react/src/App.js [hunk 2]
```diff
@@ -16,33 +21,49 @@ import {
...
         const allMethods1 = [...allMethods]
         const filteredIndex = allMethods1.findIndex(item => item.name===methodName)
         allMethods1[filteredIndex].type=methodType
         setAllMethods(allMethods1)
-        // console.log(allMethods)   
+    }
+
+    const [takenMethod, setTakenMethod] = useState('')
+    const takeMethod = (evt) => {
+      setTakenMethod(evt.target.textContent)
```
Comment (bjatta) [inline methods-react/src/App.js]: И опять

**Review by bjatta — APPROVED** [review]

File: methods-react/src/App.js (hunk 1 above)
Comment (mazurkevichandrei) [reply methods-react/src/App.js]: Fixed, button's text sets from theme:
image
image
image

File: methods-react/src/App.js (hunk 2 above)
Comment (mazurkevichandrei) [reply methods-react/src/App.js]: Fixed, method's name provided with props:
image

**Review by bjatta — APPROVED** [review]

**Pull request merged**